README.md

```diff
@@ -3,76 +3,185 @@ Use a LINQ-Enabled version of the Repository Pattern for Entity Framework
+public interface IRepository : IDisposable
+{
+
+	/// <summary>Event that fires when an item is added</summary>
+	event Action<object> ItemAdding;
+
+	/// <summary>Event that fires when an itemis modified</summary>
+	event Action<object> ItemModifing;
+
+	/// <summary>Event that fires when an item is deleted</summary>
+	event Action<object> ItemDeleting;
+
+
+	/// <summary>Queriable Entity</summary>
+	IQueryable<TEntity> Query<TEntity>() where TEntity : class, new();
+
 	/// <summary>
-	/// Interface for interacting with data storage through the repository pattern
+	/// Join another entity
 	/// </summary>
 	/// <typeparam name="TEntity"></typeparam>
-	public interface IRepository<TEntity> : IDisposable where TEntity : class, new()
-	{
-		/// <summary>Queriable Entity</summary>
-		IQueryable<TEntity> Entity { get; }
-
-		/// <summary>
-		/// Find an entity based on key(s)
-		/// </summary>
-		/// <param name="keys">The key(s) for the table</param>
-		/// <returns>Entity if found, otherwise null</returns>
-		TEntity FindOne(params object[] keys);
-
-		/// <summary>
-		/// Add or update entities
-		/// </summary>
-		/// <param name="values">Entities to add</param>
-		void AddOrUpdate(params TEntity[] values);
-
-		/// <summary>
-		/// Add or update entities
-		/// </summary>
-		/// <param name="collection">Entities to add</param>
-		void AddOrUpdate(IEnumerable<TEntity> collection);
-
-		/// <summary>
-		/// Delete a single entity by key(s)
-		/// </summary>
-		/// <param name="keys">The key(s) for the table</param>
-		void DeleteOne(params object[] keys);
-
-		/// <summary>
-		/// Delete one or more entities
-		/// </summary>
-		/// <param name="values">Entities to delete</param>
-		void Delete(params TEntity[] values);
-
-		/// <summary>
-		/// Delete one or more entities
-		/// </summary>
-		/// <param name="collection">Entities to delete</param>
-		void Delete(IEnumerable<TEntity> collection);
-
-		/// <summary>
-		/// Save pending changes for the collection
-		/// </summary>
-		/// <returns>Number of affected entities</returns>
-		int Save();
-
-		/// <summary>
-		/// Save pending changes for the collection async
-		/// </summary>
-		/// <returns>Number of affected entities</returns>
-		Task<int> SaveAsync();
-
-		/// <summary>
-		/// Save pending changes for the collection async with cancellation
-		/// </summary>
-		/// <param name="cancellationToken">Cancelation Token</param>
-		/// <returns>Number of affected entities</returns>
-		Task<int> SaveAsync(CancellationToken cancellationToken);
-
-		/// <summary>Event that fires when an item is added</summary>
-		event Action<TEntity> ItemAdded;
-
-		/// <summary>Event that fires when an itemis modified</summary>
-		event Action<TEntity> ItemModified;
-
-		/// <summary>Event that fires when an item is deleted</summary>
-		event Action<TEntity> ItemDeleted;
\ No newline at end of file
+	/// <returns></returns>
+	IQueryable<TEntity> Join<TEntity>() where TEntity : class, new();
```

Comment:
I worry that this method bakes us a little bit too much into Entity Framework. The original idea was that it could be backed by things like Raven, Mongo, or other non-EF framework. Does it also make it harder to mock? Is it even necessary?